nano/core_test/websocket.cpp

```diff
@@ -124,6 +124,54 @@ TEST (websocket, confirmation)
 	ASSERT_TIMELY (5s, future.wait_for (0s) == std::future_status::ready);
 }
 
+// Tests getting notification of a started election
+TEST (websocket, started_election)
+{
+	nano::system system;
+	nano::node_config config (nano::get_available_port (), system.logging);
+	config.websocket_config.enabled = true;
+	config.websocket_config.port = nano::get_available_port ();
+	auto node1 = system.add_node (config);
+
+	std::atomic<bool> ack_ready{ false };
+	auto task = ([&ack_ready, config, &node1] () {
+		fake_websocket_client client (node1->websocket_server->listening_port ());
+		client.send_message (R"json({"action": "subscribe", "topic": "started_election", "ack": "true"})json");
+		client.await_ack ();
+		ack_ready = true;
+		EXPECT_EQ (1, node1->websocket_server->subscriber_count (nano::websocket::topic::started_election));
+		return client.get_response ();
+	});
+	auto future = std::async (std::launch::async, task);
+
+	ASSERT_TIMELY (5s, ack_ready);
+
+	// Create election, causing a websocket message to be emitted
+	nano::keypair key1;
+	nano::block_builder builder;
+	auto send1 = builder
+				 .send ()
+				 .previous (nano::dev::genesis->hash ())
+				 .destination (key1.pub)
+				 .balance (0)
+				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
+				 .work (*system.work.generate (nano::dev::genesis->hash ()))
+				 .build_shared ();
+	nano::publish publish1{ nano::dev::network_params.network, send1 };
+	auto channel1 = node1->network.udp_channels.create (node1->network.endpoint ());
```

Comment:
There's work in progress to remove UDP so this should be using nano::transport::inproc::channel if TCP isn't needed.

Reply:
I copied that part from the the stopped_election testcase which also uses udp_channel.